lib/DiscoveryStreamFeed.jsm

```diff
@@ -40,11 +286,35 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     this._prefCache = {};
   }
 
+  replaceEndpointApiKey(url, apiKey) {
+    return url.replace("$apiKey", apiKey);
+  }
+
   finalLayoutEndpoint(url, apiKey) {
     if (url.includes("$apiKey") && !apiKey) {
       throw new Error(`Layout Endpoint - An API key was specified but none configured: ${url}`);
     }
-    return url.replace("$apiKey", apiKey);
+    return this.replaceEndpointApiKey(url, apiKey);
+  }
+
+  replaceDefaultLayoutApiKey(apiKey) {
+    if (!apiKey) {
+      throw new Error(`Default Layout - An API key was specified but none configured`);
```

Comment:
Not sure if we need to conditionally inject api key or just always try. Layout, component, spocs all go through `async fetchFromEndpoint` anyway, so unconditionally replace any $apiKey in the url with the pref's value.

Reply:
I tried to go through the pros and cons, and while I think there are some arguments for and against both ways, I couldn't come up with any strong reason one way or another, so I'm happy to make this change or not.

Reply:
OK, thinking about this, I think I agree with this change.

Reply:
I think all I needed was a comment in the replace that clears up my main con for doing it in `fetchFromEndpoint`

Reply:
It's def a lot less code.